chore: Add code formatting

dimkl opened the pull request.

gkats commented on Gemfile.lock:
```diff
@@ -1,27 +1,55 @@
   remote: .
   specs:
-    clerk-sdk-ruby (2.11.1)
+    clerk-sdk-ruby (3.0.0)
```
:question: Is this patch part of the changes of this PR?

gkats reviewed (approved):
Nice one!

How are we planning to use this, when is the formatter supposed to run?

dimkl commented:
> Nice one!
> 
> How are we planning to use this, when is the formatter supposed to run?

I think we should at least run it manually until we create GH action to run make PRs fail if there are any lint changes found. I think this is the approach we are following to rest of our projects. What do you think?

dimkl commented:
@gkats I found the following resources for rubocop linter:
- github action: added step in main.yml as part of this PR
- plugin for vim: https://github.com/ngmy/vim-rubocop
- plugin vscode: https://github.com/rubocop/vscode-rubocop
- i have also added running as a pre-commit hook with the following code: 
```ruby
require 'english'
require 'rubocop'
 
 ADDED_OR_MODIFIED = /A|AM|^M/.freeze
 
 changed_files = `git status --porcelain`.split(/\n/).
     select { |file_name_with_status|
       file_name_with_status =~ ADDED_OR_MODIFIED
     }.
     map { |file_name_with_status|
       file_name_with_status.split(' ')[1]
     }.
     select { |file_name|
       File.extname(file_name) == '.rb'
     }.join(' ')
 
 system("rubocop #{changed_files}") unless changed_files.empty?
 
 exit $CHILD_STATUS.exitstatus
 ```

gkats commented:
:clap: :clap: Nice @dimkl!!

One thing I would suggest is to remove the pre-commit hook and replace it with:
1. a script that runs the rubocop checks so that we can run it manually and
2. a CI step which errors out if rubocop checks are failing (I guess this is done)

The reason I'm suggesting we remove the pre-commit hook is that we should be allowed to commit changes without necessarily conforming to the linter in our local setup. Sometimes you have work in progress that you'd like to commit and the pre-commit hooks get in your way.

The goal would be to ensure that the code that gets merged does not violate the linter rules. What do you think?

dimkl commented:
> 1. a script that runs the rubocop checks so that we can run it manually and
I would suggest we keep the pre-commit hook as the norm would be to be compliant with our linter and adjust it to match the styles that the teams wants to use. If someone wants to by-pass the pre-commit hook they can use `git commit --no-verify ....` to bypass the hook.
 > 2.a CI step which errors out if rubocop checks are failing (I guess this is done)
 I have already added it, that's why this PR fails. I have added some TODOs in the rubocop file to loosen up the lining rules for now but i will try to resolve them as part of this PR and remove the TODOs.
 I have added it as a step in the main GH action instead of a different check.
 cc: @gkats

gkats commented:
> git commit --no-verify

That works for me! :smile: :tada:
